src/Microsoft.ML.Featurizers/CategoryImputer.cs

```diff
@@ -0,0 +1,766 @@
+    /// <seealso cref="CategoryImputerExtensionClass.ImputeCategories(TransformsCatalog, string, string)"/>
+    public sealed class CategoryImputerEstimator : IEstimator<CategoryImputerTransformer>
+    {
+        private readonly Options _options;
+
+        private readonly IHost _host;
+
+        #region Options
+        internal sealed class Column : OneToOneColumn
+        {
+            internal static Column Parse(string str)
+            {
+                Contracts.AssertNonEmpty(str);
+
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            internal bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        internal sealed class Options : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition (optional form: name:src)",
+                Name = "Column", ShortName = "col", SortOrder = 1)]
+            public Column[] Columns;
+        }
+
+        #endregion
+
+        internal CategoryImputerEstimator(IHostEnvironment env, Options options)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host.Check(!CommonExtensions.OsIsCentOS7(), "CentOS7 is not supported");
+            _host = env.Register(nameof(CategoryImputerEstimator));
+
+            foreach (var columnPair in options.Columns)
+            {
+                columnPair.Source = columnPair.Source ?? columnPair.Name;
+            }
+
+            _options = options;
+        }
+
+        public CategoryImputerTransformer Fit(IDataView input)
+        {
+            return new CategoryImputerTransformer(_host, input, _options.Columns);
+        }
+
+        public SchemaShape GetOutputSchema(SchemaShape inputSchema)
+        {
+            var columns = inputSchema.ToDictionary(x => x.Name);
+
+            foreach (var column in _options.Columns)
+            {
+                var inputColumn = columns[column.Source];
+                columns[column.Name] = new SchemaShape.Column(column.Name, inputColumn.Kind,
+                inputColumn.ItemType, inputColumn.IsKey, inputColumn.Annotations);
+            }
+
+            return new SchemaShape(columns.Values);
+        }
+    }
+
+    public sealed class CategoryImputerTransformer : RowToRowTransformerBase, IDisposable
+    {
+        #region Class data members
+
+        internal const string Summary = "Fills in missing values in a column based on the most frequent value";
+        internal const string UserName = "CategoryImputer";
+        internal const string ShortName = "CategoryImputer";
+        internal const string LoadName = "CategoryImputer";
+        internal const string LoaderSignature = "CategoryImputer";
```

Comment:
Should this be named `CategoricalImputer`?

Reply:
I am fine with either name. It was named `CategoryImputer` in the original python code, but if we think `CategoricalImputer` sounds better I am fine with renaming it.

Reply:
I'd follow the naming conventions of ML.NET (and my suggestion may be imperfect too). 

Side topic: I wouldn't perfectly replicate the existing Python code as it itself is just fulfilling a user data science need; in this case imputing categorical features with the most common value. The general goal is to fulfill the same user need; the exact methods can be a bit fungible.

Reply:
If `CategoricalImputer` would better match ML.NET naming schemes then I will update it. I am still not very good at figuring out what matches well and what doesn't.

Reply:
Renamed to CategoricalImputer